ui/components/link.js

```diff
@@ -3,24 +3,35 @@ import React from 'react';
 
 import { Link as RouterLink, routes } from '../routes';
 
-export default function Link({ children, href, params, route, ...otherProps }) {
-  if (route) {
-    return (
-      <RouterLink params={params} route={route}>
-        <a {...otherProps}>{children}</a>
-      </RouterLink>
-    );
-  } else if (href) {
-    const props = {
-      ...otherProps,
-      href,
-    };
-    if (href.search(/^https|http|^\/\//) === 0) {
-      props.target = '_blank';
-      props.rel = 'noopener noreferrer';
-      props['aria-label'] = 'Link opens in a new window';
+export default class Link extends React.Component {
+  focus() {
+    if (this.el) {
+      this.el.focus();
     }
-    return <a {...props}>{children}</a>;
+  }
+
+  render() {
+    const { children, href, params, route, ...otherProps } = this.props;
+
+    if (route) {
+      return (
+        <RouterLink params={params} route={route}>
+          <a {...otherProps}>{children}</a>
+        </RouterLink>
+      );
+    } else if (href) {
+      const props = {
+        ...otherProps,
+        href,
+      };
+      if (href.search(/^https|http|^\/\//) === 0) {
+        props.target = '_blank';
+        props.rel = 'noopener noreferrer';
+        props['aria-label'] = 'Link opens in a new window';
+      }
+      return <a {...props} ref={(el) => { this.el = el; }}>{children}</a>;
```

Comment:
Would it be clearer to move the `ref` definition up to line ~26, where we're creating `props`?

Reply:
Oh sure, I can do that. It's funny, in some ways I think of `ref` as a sort of "meta-prop" or something, so it felt OK specifying it in the JSX rather than in the `props` dict.  But I think it could also make the code more readable by not making the JSX too lengthy.